### PR TITLE
Upgrade `black` to use `--force-exclude` option for simplifying pre-commit hook

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <hkawamura0130@gmail.com>

## What changes are proposed in this pull request?

Upgrade `black` to use `--force-exclude` option for simplifying pre-commit hook:

```
# Now:
# `grep -v '^mlflow/protos'` is because `--exclude` option specified in `pyproject.toml` is ignored
# when files are passed on the command line.
black --check $(echo $git_diff | grep -v '^mlflow/protos')

# With `--force-exclude`:
# We no longer need `grep -v '^mlflow/protos'`
black --check $git_diff
```

## How is this patch tested?

NA

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Check the status of the `ci/circleci: build_doc` check. If it's successful, proceed to the
   next step, otherwise fix it.
2. Click `Details` on the right to open the job page of CircleCI.
3. Click the `Artifacts` tab.
4. Click `docs/build/html/index.html`.
5. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: hooks/pre-commit [hunk 1]
```diff
@@ -1,14 +1,36 @@
...
-  # We specify `--exclude` option in `pyproject.toml`,
-  # but it's only consulted for recursive search, not for files passed on the command line:
-  # https://github.com/psf/black/issues/438#issuecomment-413886281
-  # `--force-exclude` option can solve this issue, but black >= 20.8b0 is required:
-  # https://black.readthedocs.io/en/stable/change_log.html#id26
-  # TODO: Upgrade black to >= 20.8b0 and use `--force-exclude`
-  black --check $(echo $diff | grep -v '^mlflow/protos')
+  echo "Running black..."
+  black --check $diff
+  echo "Running pylint..."
```
Comment (harupy) [inline hooks/pre-commit:34]: We no longer need to use `grep -v '^mlflow/protos'`.

File: mlflow/store/artifact/hdfs_artifact_repo.py [hunk 2]
```diff
@@ -71,12 +71,12 @@ def log_artifacts(self, local_dir, artifact_path=None):
...
-            (self.path contains the base path - hdfs:/some/path/run_id/artifacts)
+        Lists files and directories under artifacts directory for the current run_id.
+        (self.path contains the base path - hdfs:/some/path/run_id/artifacts)
 
-            :param path: Relative source path. Possible subdirectory existing under
-                         hdfs:/some/path/run_id/artifacts
-            :return: List of FileInfos under given path
+        :param path: Relative source path. Possible subdirectory existing under
+                     hdfs:/some/path/run_id/artifacts
+        :return: List of FileInfos under given path
```
Comment (harupy) [inline mlflow/store/artifact/hdfs_artifact_repo.py:79]: Black automatically aligns docstrings.

File: pyproject.toml [hunk 3]
```diff
@@ -1,20 +1,8 @@
 [tool.black]
 line-length = 100
-# TODO: Use `--extend-exclude` once we update black version to >= 21.4b0
-exclude = '''
+target-version = ['py36']
+force-exclude = '''
 /(
-  # default exclude
```
Comment (BenWilson2) [inline pyproject.toml:6]: this new version of black has a config that ignores all of these automatically?

**Review by BenWilson2 — APPROVED** [review]
If for nothing else, the docstring formatter in this new version is great!

File: pyproject.toml (hunk 3 above)
Comment (harupy) [reply pyproject.toml:6]: Yes, in older versions of `black`, `exclude` option overwrites the default `exclude`.

Comment (harupy) [reply pyproject.toml:6]: If we specify `exclude`, it overwrites the default value of `exclude`.

**Pull request merged**